test/cql-pytest/test_cast.py

```diff
@@ -0,0 +1,165 @@
+# but it's illegal to cast a bigint to int (change 4 bytes -> 8 bytes).
+# This simplifies things, to cast we can just reinterpret the value as the other type.
+
+# Another useful use of C-style casts is type hints.
+# Sometimes it's impossible to infer the exact type of an expression from the context.
+# In such cases the type can be specified by casting the expression to this type.
+# For example: `overloadedFunction((int)?)`
+# Without the cast it's impossible to guess what should be the bind marker's type.
+# The function is overloaded, so there are many possible argument types.
+# The type hint specifies that the bind marker has type int.
+
+# An interesting thing is that such casts don't have to be explicit.
+# CQL allows to put an int value in a place where a blob value is expected
+# and it will be automatically converted without any explicit casting.
+
+
+import pytest
+from pytest import fixture
+from cassandra.protocol import InvalidRequest
+from util import unique_name, unique_key_int
+import uuid
+
+@pytest.fixture(scope="module")
+def table1(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (pk int PRIMARY KEY, blob_col blob, int_col int, bigint_col bigint)")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+@pytest.fixture(scope="module")
+def table2(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (pk int PRIMARY KEY, d date)")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+# Test casting int values to blob values using C-style casts
+def test_cast_int_blob(cql, table1):
+    pk = unique_key_int()
+
+    # Implicitly casting an integer constant to blob fails, it's unkown what the exact type is - is it tinyint, bigint?
+    # It's important that it stays this way - in the future we might implement guessing the type of untyped constants
+    # by assiging the smallest type that can fit the constant, but I think that we shouldn't allow converting them
+    # to blob without explicitly specifying size of the integer.
+    with pytest.raises(InvalidRequest, match='blob'):
+        cql.execute(f"INSERT INTO {table1} (pk, blob_col) VALUES ({pk}, 12)")
+
+    # Putting (blob) before the integer also fails, it's still unkown what the exact type is.
+    with pytest.raises(InvalidRequest, match='blob'):
+        cql.execute(f"INSERT INTO {table1} (pk, blob_col) VALUES ({pk}, (blob)123)")
+
+    # Putting (int) before the integer specifies the exact type and it's possible to cast the value.
+    # This type of assignment isn't currently allowed, will work in the future.
+    with pytest.raises(InvalidRequest, match='blob'):
+        cql.execute(f"INSERT INTO {table1} (pk, blob_col) VALUES ({pk}, (int)1234)")
+        assert list(cql.execute(f"SELECT pk, blob_col FROM {table1} WHERE pk = {pk}")) == [(pk, int(1234).to_bytes(4, 'big'))]
```

Comment:
I did it this way because I thought that it makes it a bit easier to review.
The first time these tests are introduced the reviewer can read all of the tests in the file and decide that they seem to make some sense.
Then when the tests are modified by removing `pytest.raises` there are no other changes - all of the mental load of reading the tests is during the first read, the second time around the change only removes `pytest.raises`, so there is no need for a full context switch to test reading for the reviewer.
On the other hand it might be hard to believe that a line is correct when it never gets executed.

I have no strong opinion, changed it.